src/lib/y2partitioner/actions/controllers/md.rb

```diff
@@ -278,10 +283,23 @@ def available?(partition)
         # @param partition [Y2Storage::Partition]
         # @return [Boolean] true if it is valid; false otherwise.
         def valid_for_md?(partition)
-          partition.id.is?(:linux_system) &&
-            !partition.type.is?(:extended) &&
-            partition.lvm_pv.nil? &&
-            partition.md.nil?
+          partition.id.is?(:linux_system) && !partition.type.is?(:extended)
+        end
+
+        # Whether the device is available to be used in a MD RAID
+        #
+        # A device is available if it is not already used by a RAID or
+        # used as physical volume or assigned to a mount point.
+        #
+        # @note This method does not check if the device is suitable (correct
+        # type, etc.), only if it's not already in use.
+        #
+        # @param device [Y2Storage::BlkDevice]
+        # @return [Boolean] true if can be used for a MD RAID; false otherwise.
+        def available?(device)
+          return false unless device.component_of.empty?
+
+          device.filesystem.nil? || device.filesystem.mount_point.nil?
```

Comment:
so if disk already has some filesystem, then it cannot used and at first it need somehow get rid of that filesystem, right?

Reply:
Not exactly. If the disk already has a mount point, then it cannot be used. Having a filesystem is fine as long as is not assigned to any mount point.